File: yandex_backend/24. Ошибка/solution.cs
using System;
using System.Linq;

class Program 
{
    static void Main()
    {
        int n = int.Parse(Console.ReadLine());
        List<(double, double)> servers = new List<(double, double)>();

        for (int i = 0; i < n; i++)
        {
            string[] input = Console.ReadLine().Split();
            int a = int.Parse(input[0]);
            int b = int.Parse(input[1]);
            servers.Add((a / 100.0, b / 100.0));
        }

        double totalErrorProbability = 0.0;
        foreach (var (a, b) in servers)
        {
            totalErrorProbability += a * b;
        }

        foreach (var (a, b) in servers)
        {
            double serverErrorProbability = a * b;
            double probabilityOfErrorOnServer = serverErrorProbability / totalErrorProbability;
            Console.WriteLine($"{probabilityOfErrorOnServer:F10}");
        }
    }
}

File: yandex_backend/24. Ошибка/solution_.py
import sys
from typing import List, Tuple

def main() -> None:
    n: int = int(input())
    servers: List[Tuple[float, float]] = []
    for _ in range(n):
        a, b = map(int, input().split())
        servers.append((a / 100.0, b / 100.0))

    total_error_probability: float = 0.0
    for a, b in servers:
        total_error_probability += a * b

    for a, b in servers:
        server_error_probability: float = a * b
        probability_of_error_on_server: float = server_error_probability / total_error_probability
        print(probability_of_error_on_server)

if __name__ == "__main__":
    main()